src/main/java/com/github/magickoders/JarReader.java

```diff
@@ -0,0 +1,153 @@
+         * @param signature not used here
+         * @param superName not used here
+         * @param interfaces not used here
+         */
+        @Override
+        public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+            var entity = new Entity(getModifiers(access), name.replace("/", "."), getStereotype(access), List.of(), List.of());
+            entities.add(entity);
+        }
+
+
+        /**
+         * method that returns the modifiers from the access obtained by a visit method of {@link MyVisitor}
+         *
+         * @param access the value obtained by {@link #visit(int, int, String, String, String, String[])}
+         * @return a set of modifiers which can contain private, static, public...
+         * @see Modifier
+         */
+        private static Set<Modifier> getModifiers(int access) {
+            var modifiers = new HashSet<Modifier>();
+            if (java.lang.reflect.Modifier.isPublic(access)) {
+                modifiers.add(Modifier.PUBLIC);
+            } else if (java.lang.reflect.Modifier.isPrivate(access)) {
+                modifiers.add(Modifier.PRIVATE);
+            } else if (java.lang.reflect.Modifier.isProtected(access)) {
+                modifiers.add(Modifier.PROTECTED);
+            } else {
+                modifiers.add(Modifier.PACKAGE);
+            }
+
+            if ((access & Opcodes.ACC_STATIC) != 0) {
+                modifiers.add(Modifier.STATIC);
+            }
+            // TODO à finir avec les static et sealed...
+            return Set.copyOf(modifiers);
+        }
+
+        /**
+         * method that returns the specific stereotype of a class. If it is a simple class, returns Optional.empty()
+         *
+         * @param access the value obtained by {@link #visit(int, int, String, String, String, String[])}
+         * @return an Optional which contains the stereotype of the class. Optional.empty() if it is a simple class
+         */
+        private static Optional<String> getStereotype(int access) {
+            if ((access & Opcodes.ACC_RECORD) != 0) {
+                return Optional.of("record");
+            } else if ((access & Opcodes.ACC_INTERFACE) != 0) {
+                return Optional.of("interface");
+            } else if ((access & Opcodes.ACC_ENUM) != 0) {
+                return Optional.of("enum");
+            } else if ((access & Opcodes.ACC_ANNOTATION) != 0) {
+                return Optional.of("annotation");
+            } else {
+                return Optional.empty();
+            }
+        }
+    }
+
+    /**
+     * This method returns the list of entities in the directory "target"
+     *
+     * @return the list of entities contained in the directory "target"
+     * @throws IOException
+     *
+     * @see #getEntities(String)
+     */
+    public static List<Entity> getEntities() throws IOException {
+        return getEntities("target");
```

Comment:
using the string "target" is ugly, all configuration should be taken as parameter of the constructor or the method